WordPress/Classes/Models/Blog.m

```diff
@@ -453,21 +453,23 @@ - (BOOL)jetpackAccountIsDefaultAccount
 
 - (NSNumber *)dotComID
 {
-    /*
-     mergeBlogs isn't atomic so there might be a small window for Jetpack sites
-     where self.account is the WordPress.com account, but self.blogID still has
-     the self hosted ID.
-     
-     Even if the blog is using Jetpack REST, self.jetpack.siteID should still
-     have the correct wp.com blog ID, so let's try that one first
-     */
-    if (self.jetpack.siteID) {
-        return self.jetpack.siteID;
-    } else if (self.account) {
-        return self.blogID;
-    } else {
-        return nil;
+    [self willAccessValueForKey:@"blogID"];
+    NSNumber *dotComID = [self primitiveValueForKey:@"blogID"];
+    if (dotComID.integerValue == 0) {
+        dotComID = self.jetpack.siteID;
+        if (dotComID.integerValue > 0) {
+            self.dotComID = dotComID;
+        }
     }
+    [self didAccessValueForKey:@"blogID"];
+    return dotComID;
+}
+
+- (void)setDotComID:(NSNumber *)dotComID
+{
+    [self willChangeValueForKey:@"blogID"];
+    [self setPrimitiveValue:dotComID forKey:@"blogID"];
+    [self didChangeValueForKey:@"blogID"];
```

Comment:
I'm not sure we'd gain much, considering I'm going to remove blogID once we test this in production a bit more.